### ISSUE TITLE
Treatment of Learning Rate in AdaDelta

### ISSUE THREAD

**Issue by madisonmay** [lisa-lab/pylearn2#1028]
The learning rate parameter is used in a very unintuitive way in the implementation of the `AdaDelta` learning rule -- it's scaled by the `lr_scalers` and then fed in as the epsilon parameter described in [AdaDelta: An Adaptive Learning Rate Method](http://arxiv.org/pdf/1212.5701v1.pdf).  This is not at all transparent and could lead to confusion when using `lr_scalers` or manual `learning_rate` modification does not lead to the expected result. 

 Is it worth leaving `learning_rate` as an coefficient of the update term and instead asking for a separate `epsilon` parameter?  I know the goal of `AdaDelta` is to eliminate the need for a `learning_rate` parameter / other sensitive hyperparameters (meaning that a learning rate of 1 is a good choice), but I think it's probably cleaner to keep the usage of the SGD `learning_rate` parameter consistent across learning rules and to be explicit about the use of a different type of hyperparameter in `AdaDelta`.  Thoughts?

Relevant code [here](https://github.com/lisa-lab/pylearn2/blob/master/pylearn2/training_algorithms/learning_rule.py#L238).
